common/protob/messages-cardano.proto

```diff
@@ -38,6 +38,27 @@ enum CardanoPoolRelayType {
     MULTIPLE_HOST_NAME = 2;
 }
 
+/**
+ * Auxiliary data can be sent:
+ * - directly as a BLOB - it will be included in the tx directly
+ * - with the metadata field set. The metadata object will get cbor encoded
+ *     wrapped in a TUPLE according to the Cardano CBOR spec.
+ */
+enum CardanoAuxiliaryDataType {
+  BLOB = 0;
+  TUPLE = 1;
+}
+
+/**
+ * We currently only support CATALYST_REGISTRATION. NONE is added so that
+ * we have more than one value - otherwise the enum is considered an int
+ * instead of a list and mypy complains.
+ */
+enum CardanoMetadataType {
+  NONE = 0;
```

Comment:
do we use this type for something? From CDDL it look like metadata cannot be null, only an empty map: https://github.com/input-output-hk/cardano-ledger-specs/blob/f7deb22be14d31b535f56edc3ca542c548244c67/shelley-ma/shelley-ma-test/cddl-files/shelley-ma.cddl#L213-L214

In general, if we don't need it now, I'd not bother with "NONE" type at all and I'd keep the "CATALYST_REGISTRATION" type only - it may look a bit odd, but makes sense for the sake of explicitness and it would still be less odd than a enum value which is unused/doesn't have a clear meaning (empty map vs null - both can be considered "none")

Reply:
No, we don't need it. Take a look at the comment above the declaration (which obviously isn't a good spot for it, since you missed it). If I leave the `NONE` out and get:
```
enum CardanoMetadataType {
  CATALYST_REGISTRATION = 0;
}
```
This then gets transformed into:
```
1: ('type', p.EnumType("CardanoMetadataType", (0)), p.FLAG_REQUIRED),
```
instead of (notice the `,` in the value tuple):
```
1: ('type', p.EnumType("CardanoMetadataType", (0,)), p.FLAG_REQUIRED),
```
thus making it an `int` instead of `Iterable[int]` which is expected by the `p.EnumType()` function and then `mypy` complains.

That is why I added the `NONE` - I thought it was rather harmless.

Another option might be to fix handling of a one value enum in the `pb2py.py` script directly, which it seems is written by Trezor guys themselves and it should be as simple as `if len == 1`. I'm not sure how they would like it though and whether there isn't a cleaner solution.

It's also possible to remove the enum for now, but that would mean that clients would need to be updated later when another value is added and I also like the explicitness of it as you mentioned.

Maybe moving the comment above the `NONE` value and making it more clear is also a viable solution.

Reply:
as far as I checked, no other protobuf message has a one-element enum, so changing the behavior of parsing one-element enums should not have impact beyond Cardano, for now. It seems to me that making parsing of one-element enums consistent with the rest is a change for the better, making the overall enum interpretation behavior more predictable - let's do it and raise it with Trezor devs - chances are they'd acknowledge it was just an uncovered edge-case

Reply:
Done in https://github.com/vacuumlabs/trezor-firmware/pull/57/commits/a616e3855ad94e1036f22c68d7f0fb043a08b385. This variation of the if seemed simplest and most explicit to me.